drivers/entropy/entropy_nrf5.c

```diff
@@ -8,90 +8,251 @@
 #include <entropy.h>
 #include <atomic.h>
 #include <nrf_rng.h>
+#include <soc.h>
+
+struct rand {
+	u8_t count;
+	u8_t threshold;
+	u8_t first;
+	u8_t last;
+	u8_t rand[0];
+};
+
+#define RAND_DEFINE(name, len) u8_t name[sizeof(struct rand) + len]  __aligned(4)
+
+#define RAND_THREAD_LEN (CONFIG_ENTROPY_NRF5_THR_THRESHOLD + 1)
```

Comment:
we need two more.... CONFIG_ENTROPY_NRF5_THR/ISR_SIZE, these have the "range CONFIG_ENTROPY_NRF5_THR/ISR_THRESHOLD 255"

and use (CONFIG_ENTROPY_NRF5_THR/ISR_SIZE + 1) as the context-safe ring buffer size.